Add generic/typed `take_struct` imports + ops

khagankhan opened the pull request:
- Add a function import that takes a `structref` argument to our Wasm binary skeleton
- Add a fuzzer op that calls that new function
- Add a function import for each defined struct type that takes a reference of that type specifically
- Add a fuzzer op that calls these new typed functions
- Generic `(type (;4;) (func (param (ref any))))` is called after `StructNew` op instead of `Drop` to observe that specific struct and also maintain stack discipline.
- Typed versions are called with specific structs of the types
---
cc @fitzgen @eeide

fitzgen reviewed (commented):
Looks good with the below addressed, thanks!

khagankhan commented on crates/fuzzing/src/generators/table_ops.rs:
```diff
@@ -651,8 +695,18 @@ impl TableOp {
             Self::StructNew(x) => {
-                func.instruction(&Instruction::StructNew(x + 4));
-                func.instruction(&Instruction::Drop);
+                func.instruction(&Instruction::StructNew(x + 5));
+                func.instruction(&Instruction::Call(take_structref_idx));
+            }
+            Self::TakeStructCall(x) => {
+                func.instruction(&Instruction::StructNew(x + 5));
+                func.instruction(&Instruction::Call(take_structref_idx));
+            }
+            Self::TakeTypedStructCall(x) => {
+                let s = STRUCT_BASE + x;
+                let f = TYPED_FN_BASE + x;
+                func.instruction(&Instruction::StructNew(s));
+                func.instruction(&Instruction::Call(f));
```
Yes, I agree!

khagankhan commented:
Thanks @fitzgen. Working on them!

khagankhan commented:
@fitzgen @eeide
Ready for review 
Changes
- Split `table_ops.rs` into separate  modules and renamed to `GcOps` where applicable
- Addressed prior review feedback
- Ran the fuzzer for ~1 hour and fixed generation bugs discovered
